presto-spi/src/main/java/com/facebook/presto/spi/type/FloatType.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.type;
+
+import com.facebook.presto.spi.ConnectorSession;
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockBuilder;
+
+import static com.facebook.presto.spi.StandardErrorCode.NUMERIC_VALUE_OUT_OF_RANGE;
+import static com.facebook.presto.spi.type.TypeSignature.parseTypeSignature;
+import static io.airlift.slice.SizeOf.SIZE_OF_FLOAT;
+
+public final class FloatType
+        extends AbstractFixedWidthType
+{
+    public static final FloatType FLOAT = new FloatType();
+
+    private FloatType()
+    {
+        super(parseTypeSignature(StandardTypes.FLOAT), long.class, SIZE_OF_FLOAT);
+    }
+
+    @Override
+    public long getLong(Block block, int position)
+    {
+        return (long) block.getInt(position, 0);
+    }
+
+    public void writeFloat(BlockBuilder blockBuilder, float value)
+    {
+        writeLong(blockBuilder, Float.floatToRawIntBits(value));
+    }
+
+    @Override
+    public void writeLong(BlockBuilder blockBuilder, long value)
+    {
+        try {
+            Math.toIntExact(value);
+        }
+        catch (ArithmeticException e) {
+            throw new PrestoException(NUMERIC_VALUE_OUT_OF_RANGE, "More than 32b value provided for Float type");
+        }
+        blockBuilder.writeInt((int) value).closeEntry();
+    }
+
+    @Override
+    public Object getObjectValue(ConnectorSession session, Block block, int position)
+    {
+        if (block.isNull(position)) {
+            return null;
+        }
+        return block.getFloat(position, 0);
+    }
+
+    @Override
+    public void appendTo(Block block, int position, BlockBuilder blockBuilder)
+    {
+        if (block.isNull(position)) {
+            blockBuilder.appendNull();
+        }
+        else {
+            blockBuilder.writeFloat(block.getFloat(position, 0)).closeEntry();
+        }
+    }
+
+    @Override
+    public boolean isComparable()
+    {
+        return false;
+    }
+
+    @Override
+    public boolean isOrderable()
+    {
+        return false;
+    }
+
+    @Override
+    public boolean equalTo(Block leftBlock, int leftPosition, Block rightBlock, int rightPosition)
+    {
+        int leftValue = leftBlock.getInt(leftPosition, 0);
```

Comment:
Good finding, 0.0 == -0.0. I even put it in operator tests but forgot about this bit here.

Again, any idea how to test this part of code?

Fixed in 4fcafccbc00708417cd87ae3e50032f127eb5a84.